app/src/main/java/com/simformsolutions/sample/app/ui/main/MainViewModel.kt

```diff
@@ -34,7 +38,21 @@ class MainViewModel @Inject constructor(
     private val mainRepository: MainRepository
 ) : ViewModel() {
 
-    val repositories = Pager(
+    private val _isRefreshing = MutableStateFlow(false)
+    val isRefreshing = _isRefreshing.asStateFlow()
+
+    var repositories = getAllRepositories()
```

Comment:
Is this really a good way to refresh pager ?

Reply:
There is an alternate to this, we can use `refresh()` on LazyPagingItems<T> object to manually refresh the page. This method conflicts with our current implementation. We are using dependency injection by which we are providing `SimformRepositoriesSource` as a singleton component to `Pager` in `MainViewModel` which leads to crash into application as it requires a new instance on every refresh. You can find the error message below:

> An instance of PagingSource was re-used when Pager expected to create a new instance. Ensure that the pagingSourceFactory passed to Pager always returns a new instance of PagingSource

Due to that reason I opted for manually fetching the list again. Let me know if you are aware of any other method than this.

Reply:
I see, I was not aware about that. In this case we should not be injecting `simformRepositoriesSource` into `MainRepository` instead create instance using creator pattern. Also remove `@Inject` from the `SimformRepositoriesSource`.

Reply:
Updated